packages/start/src/index.ts

```diff
@@ -0,0 +1,31 @@
+import { createSignal, createEffect, Setter, Accessor } from "solid-js";
+import { isServer } from "solid-js/web";
+import { parseCookie } from "solid-start";
+import { useRequest } from "solid-start/server";
+
+/**
+ * A template example of how to create a new primitive.
+ *
+ * @param cookieMaxAge The max age of the cookie to be set, in seconds
+ * @param cookieName The name of the cookie to be set
+ * @return Returns an accessor and setter to manage the user's current theme
+ */
+type UserThemeOptions = {
+  cookieMaxAge: Number;
+  cookieName: String;
+};
+export const createUserTheme = <T extends String>(
+  defaultTheme: T,
+  options?: UserThemeOptions,
+): [get: Accessor<T>, set: Setter<T>] => {
```

Comment:
```suggestion
export const createUserTheme = (
  defaultTheme: string,
  options?: UserThemeOptions,
): [get: Accessor<string>, set: Setter<string>] => {
```
Can we be sure that the value received from a cookie will always be what we specified in the generic? No clue how it works, I guess we control both sides, but it feels like something that should be typed vaguely as `string` so that we force users to validate it. Or require a validation function in the options to be able to specify a generic.
But maybe it's fine, not sure